ui/imports/shared/views/AssetsView.qml

```diff
@@ -71,34 +71,20 @@ ColumnLayout {
             d.controller.settingsHideCommunityTokens(communityId, tokenSymbols)
         }
 
-        readonly property SubmodelProxyModel assetsWithFilteredBalances: SubmodelProxyModel {
-            id: assetsWithFilteredBalances
-            sourceModel: root.assets
-            submodelRoleName: "balances"
-            delegateModel: SortFilterProxyModel {
-                sourceModel: submodel
-                filters: FastExpressionFilter {
-                    expression: {
-                        root.networkFilters
-                        root.addressFilters
-                        return root.networkFilters.split(":").includes(model.chainId+"") &&
-                                root.addressFilters.split(":").includes(model.account)
-                    }
-                    expectedRoles: ["chainId", "account"]
-                }
-            }
-        }
-
         function getTotalBalance(balances, decimals) {
             let totalBalance = 0
             for(let i=0; i<balances.count; i++) {
-                totalBalance+=SQUtils.AmountsArithmetic.toNumber(ModelUtils.get(balances, i, "balance"), decimals)
+                let balancePerAddressPerChain = ModelUtils.get(balances, i)
+                if (root.networkFilters.split(":").includes(balancePerAddressPerChain.chainId+"") &&
```

Comment:
A suggestion, move this `root.networkFilters.split(":")` and `root.addressFilters.split(":")` out of the loop? You don't have to do the same thing over and over in every iteration

Reply:
done @caybro